File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.js
// 발음을 할 수 있는 단어 배열에 먼저 저장
//babbling을 for문 돌리면서 차례대로 발음할 수 있는지를 count변수에 체크
//조건 !! 반복해서 발음 불가 !!! 

function solution(babbling) {
  let babyWords = ["aya", "ye", "woo", "ma"]
  
  let cnt = 0
  for (let i=0; i<babbling.length; i++) {
    let tmpChk = babbling[i]
    //tmpchk = yeye 
    for (let j=0; j<4; j++) {
      // update - 4가지 발음에 따라 체크
      if (tmpChk.includes(babyWords[j].repeat(2))) break // 연속이면 나오기
      tmpChk = tmpChk.split(babyWords[j]).join(' ') // 'yeaya' = ['','aya'].join('')='aya' <- 순서상관없이 가능
    }
    tmpChk = tmpChk.split(' ').join('')
    if (tmpChk.length === 0) cnt ++
  }

  return cnt
}
// ['a','b','a','d'].join('%')= 'a%b%a%d'
// .join('')='abad'
// .join(' ') = 'a b a d'

//'yayae' -> split(babyWords[j]) 후 그냥 빈문자열로 join하면 안되는 이유 
// 'y', 'e' - > 'y e'

// 'yeaya' -> 'ye',''


console.log(solution(["aya", "yee", "u", "maa"]));//1
console.log(solution(["ayaye", "uuu", "yeye", "yemawoo", "ayaayaa"]))//2

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
# 그냥 옹알이 단어 앞에서부터 words에 있는 단어 차례대로 돌리면서 찾고, - index함수
# index함수의 결과로 0이 나오면, -> replace(word,'',1)
    # 또한, 여기서 한번 더 index(word)해서 값이 0이 바로 나오면 연속발음 -> break
# 최종적으로, 옹알이 단어 길이가 0이되면 cnt += 1, 아니면 넘어가고

def solution(babbling):
    words = ["aya", "ye", "woo", "ma"]
    cnt = 0

    for bab in babbling:
        tmp = 0
        flag = True

        while tmp != len(bab):
            tmp = len(bab)
            for word in words:
                if bab.find(word) == 0:
                    bab = bab.replace(word,'',1)
                    # 연속인 경우
                    if bab.find(word) == 0:
                        flag = False # flag가 False면 절대 cnt+1하지 않는다
            
        if flag == False:
            continue
        if len(bab) == 0:
            cnt += 1

    return cnt

print(solution(["aya", "yee", "u", "maa"]))
print(solution(["ayaye", "uuu", "yeye", "yemawoo", "ayaayaa"]))